File: HotDesk_SPICE/App_Code/Binding.vb
Imports Library.Root.Object
Imports System.Collections.Generic
Namespace Control
    ''' <summary>
    ''' Component Binding 
    ''' ----------------------------------------------
    ''' C.C.Yeon    16 April 2012   initial version
    ''' </summary>
    ''' <remarks></remarks>
    Public Class Binding
        Public Shared Sub BindDropDownListResource(ByVal DDL As System.Web.UI.WebControls.DropDownList, ByVal ResourceName As String, Optional ByVal Text As String = "", Optional ByVal Value As String = "")
            DDL.DataSource = Library.Root.Control.Convertion(Of Binder).Deserializer(HttpContext.GetGlobalResourceObject("SearchSource", ResourceName))
            DDL.DataTextField = "Text"
            DDL.DataValueField = "Value"
            DDL.DataBind()
            AddList(DDL, Text, Value)
        End Sub

        Public Shared Sub BindDropDownList(ByVal DDL As System.Web.UI.WebControls.DropDownList, ByVal list As List(Of Binder), Optional ByVal Text As String = "", Optional ByVal Value As String = "")
            If list.Count > 0 Then
                DDL.DataSource = list
                DDL.DataTextField = "Text"
                DDL.DataValueField = "Value"
                DDL.DataBind()
            End If
            AddList(DDL, Text, Value)
        End Sub

        Private Shared Sub AddList(ByVal ddl As System.Web.UI.WebControls.DropDownList, ByVal Text As String, ByVal Value As String)
            If Value <> String.Empty Then
                ddl.Items.Insert(0, New System.Web.UI.WebControls.ListItem(Text, Value))
            End If
        End Sub
    End Class
End Namespace

File: HotDesk_SPICE/App_Code/Binding.cs
using Library.Root.Object;
using System.Collections.Generic;
using System.Web;

namespace Control
{
    /// <summary>
    /// Component Binding 
    /// ----------------------------------------------
    /// C.C.Yeon    16 April 2012   initial version
    /// </summary>
    public class Binding
    {
        public static void BindDropDownListResource(System.Web.UI.WebControls.DropDownList DDL, string ResourceName, string Text = "", string Value = "")
        {
            DDL.DataSource = Library.Root.Control.Convertion<Binder>.Deserializer(HttpContext.GetGlobalResourceObject("SearchSource", ResourceName));
            DDL.DataTextField = "Text";
            DDL.DataValueField = "Value";
            DDL.DataBind();
            AddList(DDL, Text, Value);
        }

        public static void BindDropDownList(System.Web.UI.WebControls.DropDownList DDL, List<Binder> list, string Text = "", string Value = "")
        {
            if (list.Count > 0)
            {
                DDL.DataSource = list;
                DDL.DataTextField = "Text";
                DDL.DataValueField = "Value";
                DDL.DataBind();
            }
            AddList(DDL, Text, Value);
        }

        private static void AddList(System.Web.UI.WebControls.DropDownList ddl, string Text, string Value)
        {
            if (Value != string.Empty)
            {
                ddl.Items.Insert(0, new System.Web.UI.WebControls.ListItem(Text, Value));
            }
        }
    }
}
